src/uwtools/drivers/esg_grid.py

```diff
@@ -50,6 +50,7 @@ def namelist_file(self):
             config_class=NMLConfig,
             config_values=self._driver_config["namelist"],
             path=path,
+            schema=self._namelist_schema(schema_keys=["$defs", "namelist_content"]),
```

Comment:
Because the `esg_grid` driver has some complexity about how its namelist config is validated, it needs to provide a custom keypath to locate the subschema to use.